executions/graphql-kotlin-transaction-batcher-instrumentation/src/main/kotlin/com/expediagroup/graphql/transactionbatcher/instrumentation/state/ExecutionState.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2022 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.transactionbatcher.instrumentation.state
+
+import graphql.schema.DataFetcher
+
+enum class LevelState { NOT_DISPATCHED, DISPATCHED }
+
+/**
+ * Handle the state of an [graphql.ExecutionInput]
+ */
+class ExecutionState(documentHeight: Int) {
+
+    private val levelsState: MutableMap<Level, LevelState> = mutableMapOf(
+        *Array(documentHeight) { number -> Level(number + 1) to LevelState.NOT_DISPATCHED }
+    )
+
+    private val expectedFetches: MutableMap<Level, Int> = mutableMapOf(
+        *Array(documentHeight) { number -> Level(number + 1) to 0 }
+    )
+    private val happenedFetches: MutableMap<Level, Int> = mutableMapOf(
```

Comment:
`happened` seems kind of weird -> maybe we should rename to `completed` or something along those lines instead?

Reply:
in the GraphQLContext `completed` indicated completion stage of a fetch - this is when a completableFuture completes the async work, so maybe we could use a word related to dispatched

something like `fetchersToDispatch` and `dispatchedFetchers` ?

Reply:
maybe 
* `happenedFetches` -> `fetched`
* `happenedExecutionStrategies` -> `executedStrategies`
* `happenedOnFieldValueInfos` -> `onFieldValueInfos`

or you could keep the `dispatched` prefix